testing/upgradetest/upgrader.go

```diff
@@ -325,7 +328,15 @@ func PerformUpgrade(
 
 	upgradeOutput, err := startFixture.Exec(ctx, upgradeCmdArgs)
 	if err != nil {
-		return fmt.Errorf("failed to start agent upgrade to version %q: %w\n%s", endVersionInfo.Binary.Version, err, upgradeOutput)
+		s, ok := status.FromError(err)
```

Comment:
Does this really work here? The output from `startFixture.Exec` is an error from a command output and not directly from the GRPC client as the code above is.

Reply:
Wow really from string output? That is surprising, but awesome if that works!

Reply:
Fixed in https://github.com/elastic/elastic-agent/pull/4519/commits/1b17b6da507e7b170e9686f9be14b012d21f19e1

Reply:
Added a test in https://github.com/elastic/elastic-agent/pull/4519/commits/5cb86cd6016a9e100705f8b493ea9cf796b55b22